src/gui/editors/PianoRoll.cpp

```diff
@@ -3261,8 +3287,22 @@ int PianoRoll::getKey(int y ) const
 	return key_num;
 }
 
+QList<int> PianoRoll::getAllOctavesForKey( int keyToMirror ) const
+{
+	QList<int> keys;
 
+	for (int i = 0; i < NumKeys; ++i)
+	{
+		if ((i < keyToMirror && (keyToMirror - i) % KeysPerOctave == 0) || (i > keyToMirror && (i - keyToMirror) % KeysPerOctave == 0))
```

Comment:
@Wallacoloo nope, that second approach is definitely much more succinct and accomplishes the same thing-- thanks!  I'll fix that now.